File: classes/parte-8-como-funcionan-los-lenguajes/131-recoleccion-de-basura-gc/implementaciones/javascript/main.mjs
import { readFileSync } from "node:fs";

const n = parseInt(readFileSync(0, "utf8").trim(), 10);
for (let i = 0; i < n; i++) {
  const tmp = {}; // sin referencia persistente, será recolectado
  void tmp;
}
console.log(`creados=${n} estado=recolectado`);

File: classes/parte-8-como-funcionan-los-lenguajes/131-recoleccion-de-basura-gc/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))
	for i := 0; i < n; i++ {
		tmp := new(int) // sin referencia persistente, el GC lo recolecta
		_ = tmp
	}
	fmt.Printf("creados=%d estado=recolectado\n", n)
}
